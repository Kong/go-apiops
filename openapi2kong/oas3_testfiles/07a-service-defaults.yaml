# Some of the generated fields for services should be overridable
# using the defaults.
# These are the order in which we should take the values:
# - service.protocol: x-kong-service-defaults -> servers[1].url (first entry servers block) -> default value "https"
# - service.port: x-kong-service-defaults -> servers[1].url (first entry servers block) -> default to 80/443 based on protocol
# - service.path: x-kong-service-defaults -> servers[1].url (first entry servers block) -> default to "/"
# - service.host: x-kong-service-defaults -> if 1 url then servers[1].url (first entry servers block), if multiple create an upstream -> default to "localhost"
#   Note: if a host default is specified, then no upstream should be generated

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/anything
  - url: https://server2.com/anything

x-kong-service-defaults:
  protocol: http
  port: 123
  path: /something
  host: myhost.com

paths:
  /path1:
    get:
      operationId: uses-doc-service
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
