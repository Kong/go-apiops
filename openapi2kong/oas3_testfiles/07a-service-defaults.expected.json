{
  "_format_version": "3.0",
  "services": [
    {
      "host": "myhost.com",
      "id": "e3a95664-95d7-5b10-bc3d-11c9cc6c2b4e",
      "name": "Simple_API_overview",
      "path": "/something",
      "plugins": [],
      "port": 123,
      "protocol": "http",
      "routes": [
        {
          "id": "5f917263-5851-5850-923f-06a35814bb61",
          "methods": [
            "GET"
          ],
          "name": "Simple_API_overview-uses-doc-service",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_07a-service-defaults.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_07a-service-defaults.yaml"
      ]
    }
  ],
  "upstreams": []
}